Utilities/sysinfo.h

```diff
@@ -40,6 +40,15 @@ namespace utils
 	bool has_512();
 
 	bool has_xop();
+	
+	int is_haswell_or_broadwell();
+	
+	enum haswell_broadwell
```

Comment:
There is really no need to have an enum for this. Just make seperate check methods for haswell and broadwell